wallet-api/src/Wallet/Emulator/Types.hs

```diff
@@ -164,13 +163,37 @@ instance WalletAPI EmulatedWalletApi where
 
 -- Emulator code
 
-type Assertion = EmulatorState -> Maybe AssertionError
+data Assertion
+  = IsValidated Tx
+  | OwnFundsEqual Wallet Value
+
 newtype AssertionError = AssertionError T.Text
     deriving Show
 
-isValidated :: Tx -> Assertion
-isValidated txn emState =
-    if notElem txn (join $ emChain emState) then Just $ AssertionError $ "Txn not validated: " <> T.pack (show txn) else Nothing
+assert :: (MonadEmulator m) => Assertion -> m ()
+assert (IsValidated txn) = isValidated txn
```

Comment:
This seems to be throwing off the `stylish-haskell` CI.

Reply:
seems to be a duplicate.

Reply:
sorry, forget that, github broken UI mixing things up, this should also be fixed now